extensions/search-npm/src/index.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  ActionPanel,
+  CopyToClipboardAction,
+  List,
+  OpenInBrowserAction,
+  PushAction,
+  showToast,
+  ToastStyle,
+  Detail,
+  Icon,
+} from '@raycast/api'
+import { useEffect, useState } from 'react'
+import fetch from 'node-fetch'
+import { FetchResponse, Result } from './npmsResponse.model'
+import { fetchReadme } from 'fetch-readme'
+import parsedRepoUrl from 'parse-github-url'
+
+export default function PackageList() {
+  const [results, setResults] = useState<Result[]>([])
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const onSearchTextChange = async (text: string) => {
+    setLoading(true)
+    const response = await fetchPackages(text.replace(/\s/g, '+'))
+    setResults(response)
+    setLoading(false)
+  }
+
+  return (
+    <List
+      isLoading={loading}
+      searchBarPlaceholder="Search npms.io..."
+      onSearchTextChange={onSearchTextChange}
+      throttle
+    >
+      {results?.length ? (
+        results.map((result) => {
+          if (!result.package.links.repository) {
+            return null
+          }
+          return <PackageListItem key={result.package.name} result={result} />
+        })
+      ) : (
+        <List.Item id="Empty" title="No results" icon={Icon.XmarkCircle} />
+      )}
+    </List>
+  )
+}
+
+interface PackageListItemProps {
+  result: Result
+}
+const PackageListItem = ({
+  result,
+}: PackageListItemProps): JSX.Element | null => {
+  const pkg = result.package
+  const isGithubRepo = pkg.links.repository.includes('github')
+  const parsedRepo = parsedRepoUrl(pkg.links.repository)
+
+  return (
+    <List.Item
+      id={pkg.name}
+      key={pkg.name}
+      title={pkg.name}
+      subtitle={pkg.description}
+      icon={Icon.ArrowRight}
+      accessoryTitle={`v${pkg.version}`}
+      keywords={pkg.keywords}
+      actions={
+        <ActionPanel>
+          <OpenInBrowserAction url={pkg.links.repository} title="Repository" />
+          <OpenInBrowserAction
+            url={`https://bundlephobia.com/package/${pkg.name}`}
+            title="Bundlephobia cost"
+            icon={Icon.QuestionMark}
+          />
+          {isGithubRepo && parsedRepo?.owner && parsedRepo?.name ? (
+            <PushAction
+              title="View readme"
+              target={<Readme user={parsedRepo.owner} repo={parsedRepo.name} />}
+              icon={Icon.Eye}
+            />
+          ) : null}
+          {pkg.links.homepage !== pkg.links.repository ? (
+            <OpenInBrowserAction url={pkg.links.homepage} title="Homepage" />
+          ) : null}
+          <OpenInBrowserAction url={pkg.links.npm} title="npm" />
+          <OpenInBrowserAction
+            url={`https://snyk.io/vuln/npm:${pkg.name}`}
+            title="Snyk vulnerability check"
```

Comment:
Maybe you can add a Snyk icon to this action? Same for the other bits. 

```suggestion
            title="Open Snyk Vulnerability Check"
```

Reply:
I tried adding Snyk's icon and it is too complex to look good at this small size. Here's the best version that I could find:

![image](https://user-images.githubusercontent.com/64883/137916595-a59c018b-e77f-40c4-8e1c-79f0b4ac2956.png)

Reply:
They have a Press Kit with a monochrome version that may look good

https://snyk.io/press-kit/

Reply:
Thanks, but I tried it and it still looked awful. I have committed the version in the screenshot above for now..